packages/zosfiles/src/methods/list/List.ts

```diff
@@ -426,11 +426,18 @@ export class List {
         ImperativeExpect.toNotBeEqual(patterns.length, 0, ZosFilesMessages.missingPatterns.message);
         const zosmfResponses: IZosmfListResponse[] = [];
 
+        const maxLength = options.maxLength;
+
+        let totalCount = 0;
         // Get names of all data sets
         for (const pattern of patterns) {
+            if (maxLength && totalCount >= options.maxLength) {
+                break;
+            }
             let response: any;
             try {
-                response = await List.dataSet(session, pattern, { attributes: true, maxLength: options.maxLength, start: options.start });
+                response = await List.dataSet(session, pattern,
```

Comment:
I think it would be nice for readability to explain this a little with a comment like you did in your pr explanation: 'We'll gradually increment the total number of items returned for each pattern, and once we've reached the max length, we'll stop listing patterns.'

Reply:
I agree, thanks for the suggestion 👍 I added a few comments where that processing was added for clarification.

Addressed in 53b8b37